src/packet.cpp

```diff
@@ -38,12 +39,33 @@ Packet::Packet(const std::array<uint8_t, 4096>& m_recvBuffer, size_t bytes)
 
         attributeIndex += attributeLength;
     }
+
+    bool eapMessage = false;
+    bool messageAuthenticator = false;
+    for (size_t i = 0; i < m_attributes.size(); ++i)
+    {
+        if (m_attributes[i]->type() == EAP_MESSAGE)
+            eapMessage = true;
+
+        if (m_attributes[i]->type() == MESSAGE_AUTHENTICATOR)
+            messageAuthenticator = true;
+    }
+    try
```

Comment:
The purpose of this construction is unclear to me:
```c++
try
{
    if (something)
        throw error(...);
}
catch (const error& e)
{
    std::cout << e.what();
}
```

You throw an exception and immediately handle it with just printing the message. If you want to just print the message, why don't you do it directly and involve exception mechanism instead?

Also, it seems to me that throwing the exception out of the constructor makes more sense. We can not construct a packet which requires 'Message-Authenticator' but does not have it. In this case, I don't understand why you handle it right away and, generaly speacking, ignore it (the `Packet` is constructed regardless).